rollup.config.js

```diff
@@ -0,0 +1,252 @@
+import commonjs from 'rollup-plugin-commonjs';
+import babel from 'rollup-plugin-babel';
+import json from 'rollup-plugin-json';
+import autoExternal from 'rollup-plugin-auto-external';
+import cleanup from 'rollup-plugin-cleanup';
+import nodeGlobals from 'rollup-plugin-node-globals';
+import builtins from 'rollup-plugin-node-builtins';
+import {terser} from 'rollup-plugin-terser';
+import bundleSize from 'rollup-plugin-bundle-size';
+import resolve from 'rollup-plugin-node-resolve';
+
+const DEFAULT_INPUT = 'src/index.js';
+const NODE_TARGET = '8';
+const BROWSER_TARGET = 'last 2 versions';
+
+/**
+ * Returns the rollup configuration with globals and names set.
+ *
+ * @param {String} name
+ * @param {String} outputFileName
+ * @param {Object} globals
+ * @param {Array} dedupe
+ * @param {boolean} namedExports
+ *
+ * @returns {Array}
+ */
+export default (name, outputFileName, globals, dedupe, namedExports) => {
+    let mappedConfig = [];
+
+    // CJS
+    if (process.env.CJS === 'true') {
+        mappedConfig.push(
+            rollupConfig(
+                DEFAULT_INPUT,
+                'dist/' + outputFileName + '.cjs.js',
+                name,
+                'cjs',
+                namedExports ? 'named' : 'auto',
+                {},
+                {
+                    exclude: 'node_modules/**',
+                    targets: {
+                        node: NODE_TARGET
+                    }
+                }
+            )
+        );
+    }
+
+    // ESM
+    if (process.env.ESM === 'true') {
+        mappedConfig.push(
+            rollupConfig(
+                DEFAULT_INPUT,
+                'dist/' + outputFileName + '.esm.js',
+                name,
+                'es',
+                'auto',
+                {},
+                {
+                    exclude: 'node_modules/**',
+                    targets: {
+                        node: NODE_TARGET
+                    }
+                }
+            )
+        );
+    }
+
+    // Minified UMD
+    if (process.env.MIN === 'true') {
+        mappedConfig.push(
+            rollupConfig(
+                'dist/' + outputFileName + '.esm.js',
```

Comment:
The minified version is built from the ESM one. Does this work correctly? I'm seen rollup break `crypto-browserify` when using ESM as output.

Reply:
This does work as expected in my point of view. What exactly do you mean with break?

Reply:
Rollup CJS plugin can transform CJS modules into ESM (which is incredibly useful). I've seen this fail when using `rollup-plugin-node-builtins` with `{crypto: true}`.

Reply:
> Rollup CJS plugin can transform CJS modules into ESM (which is incredibly useful).

Oh yes, this is actually really useful! 

> I've seen this fail when using rollup-plugin-node-builtins with {crypto: true}

The builtins plugin is only used for the minified UMD bundle we will use in the browser. The native crypto dependency will be handled as an external dependency for the CJS and ESM bundle which is the correct behavior. This way does the developer has the opportunity to overwrite the native ``crypto`` package with a custom resolver rule and for the minified files we provide for ``unpkg`` and ``jsdelivr`` will it use the already in usage ``crypto-browserify`` package.

Reply:
Great! So nothing to worry about the CJS and ESM bundles. I'll keep this open for now as I continue to dig into the rollup config and understand what gets into the UMD package.